src/dotnet/Telemetry.cs

```diff
@@ -146,5 +155,49 @@ private void TrackEventTask(string eventName, IDictionary<string, string> proper
                 return _commonProperties;
             }
         }
+        
+        private List<string> GetMacs()
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            var macs = new List<string>();
+            
+            if (nics == null || nics.Length < 1)
+            {
+                macs.Add(string.Empty);
+                return macs;
+            }
+            
+            foreach (NetworkInterface adapter in nics)
+            {
+                IPInterfaceProperties properties = adapter.GetIPProperties();
+
+                PhysicalAddress address = adapter.GetPhysicalAddress();
+                byte[] bytes = address.GetAddressBytes();
+                macs.Add(string.Join("-", bytes.Select(x => x.ToString("X2"))));
```

Comment:
Why would we care if the string is in a specific format?  It is hashed, so it should be 100% opaque to us.

IMO - it should look like this:

```
private List<byte[]> GetMacs()
{
    ... same as above only add address.GetAddressBytes() to the 'macs' list
}

       private List<string> HashSha256(List<byte[]> values)
        {
            var sha256 = SHA256.Create();
            var hashedStrings = new List<string>();
            foreach (var value in values)
            {
                byte[] hash = sha256.ComputeHash(value);
                string hashString = string.Empty;
                foreach (byte x in hash)
                {
                    hashString += String.Format("{0:x2}", x);
                }
                hashedStrings.Add(hashString);
            }
            return hashedStrings;
        }
```